config-model/src/main/java/com/yahoo/schema/parser/ConvertParsedRanking.java

```diff
@@ -34,84 +34,74 @@ private RankProfile makeRankProfile(Schema schema, String name) {
     }
 
     void convertRankProfile(Schema schema, ParsedRankProfile parsed) {
-        RankProfile profile = makeRankProfile(schema, parsed.name());
-        for (String name : parsed.getInherited())
-            profile.inherit(name);
-
-        parsed.isStrict().ifPresent(profile::setStrict);
-        parsed.isUseSignificanceModel().ifPresent(profile::setUseSignificanceModel);
-
-        for (var constant : parsed.getConstants().values())
-            profile.add(constant);
-
-        for (var onnxModel : parsed.getOnnxModels())
-            profile.add(onnxModel);
-
-        for (var input : parsed.getInputs().entrySet())
-            profile.addInput(input.getKey(), input.getValue());
-
-        for (var func : parsed.getFunctions()) {
-            String name = func.name();
-            List<String> parameters = func.getParameters();
-            String expression = func.getExpression();
-            boolean inline = func.getInline();
-            profile.addFunction(name, parameters, expression, inline);
-        }
-
-        parsed.getRankScoreDropLimit().ifPresent(profile::setRankScoreDropLimit);
-        parsed.getSecondPhaseRankScoreDropLimit().ifPresent(profile::setSecondPhaseRankScoreDropLimit);
-        parsed.getTermwiseLimit().ifPresent(profile::setTermwiseLimit);
-        parsed.getPostFilterThreshold().ifPresent(profile::setPostFilterThreshold);
-        parsed.getApproximateThreshold().ifPresent(profile::setApproximateThreshold);
-        parsed.getTargetHitsMaxAdjustmentFactor().ifPresent(profile::setTargetHitsMaxAdjustmentFactor);
-        parsed.getKeepRankCount().ifPresent(profile::setKeepRankCount);
-        parsed.getMinHitsPerThread().ifPresent(profile::setMinHitsPerThread);
-        parsed.getNumSearchPartitions().ifPresent(profile::setNumSearchPartitions);
-        parsed.getNumThreadsPerSearch().ifPresent(profile::setNumThreadsPerSearch);
-        parsed.getReRankCount().ifPresent(profile::setRerankCount);
-
-        parsed.getMatchPhase().ifPresent(profile::setMatchPhase);
-        parsed.getDiversity().ifPresent(profile::setDiversity);
-
-        parsed.getFirstPhaseExpression().ifPresent(profile::setFirstPhaseRanking);
-        parsed.getSecondPhaseExpression().ifPresent(profile::setSecondPhaseRanking);
-
-        parsed.getGlobalPhaseExpression().ifPresent(profile::setGlobalPhaseRanking);
-        parsed.getGlobalPhaseRerankCount().ifPresent(profile::setGlobalPhaseRerankCount);
-
-        for (var value : parsed.getMatchFeatures()) {
-            profile.addMatchFeatures(value);
-        }
-        for (var value : parsed.getRankFeatures()) {
-            profile.addRankFeatures(value);
+        try {
+            RankProfile profile = makeRankProfile(schema, parsed.name());
```

Comment:
Perhaps separate conversion and error handling with a private conversion function.
Less indentation, and error handling is easier to follow.

```
private RankProfile createRankProfile(Schema schema, ParsedRankProfile parsed) {
    RankProfile profile = makeRankProfile(schema, parsed.name());
    ......
    return profile;
}

void convertRankProfile(Schema schema, ParsedRankProfile parsed) {
    try {
        rankProfileRegistry.add(createRankProfile(schema, parsed));
    } catch (IllegalArgumentException e) {
         throw new IllegalArgumentException("In " + parsed, e);
    }
}
```

Reply:
Okay